server/src/main/resources/rules/rules.js

```diff
@@ -188,22 +201,51 @@ function createPool(pool) {
         if (attr_result === null) {
             return this.findAttributeIn(attrName, this.productAttributes);
         }
-        return attr_result;
-    };
 
-    pool.hasAttribute = function(attrName) {
-        return this.getAttribute(attrName) !== null;
+        return attr_result;
     };
 
     pool.getProductAttribute = function (attrName) {
         var attr_result = this.findAttributeIn(attrName, this.productAttributes);
         if (attr_result === null) {
             return this.findAttributeIn(attrName, this.attributes);
         }
+
         return attr_result;
     };
 
+    pool.attributeIsPresent = function(attributes, attributeName) {
+        if (attributes !== undefined) {
+            if (Array.isArray(attributes)) {
+                // Old-style array of attribute objects
+                for (var k = 0; k < attributes.length; k++) {
+                    if (attributeName == attributes[k].name) {
+                        return true;
+                    }
+                }
+            }
+            else if (attributes.hasOwnProperty(attributeName)) {
+                // Attributes stored as a map
+                return true;
+            }
+        }
+
+        return false;
+    };
+
+    pool.hasAttribute = function(attrName) {
+        // TODO: This isn't accurate. We consider an attribute to be present if it's set, even if
+        // it has a null value. We should update this method accordingly.
+        // return this.attributeIsPresent(this.attributes, attrName) || this.attributeIsPresent(this.productAttributes, attrName);
```

Comment:
Nah. The behavior is still incorrect here, and the comment just shows what the correct behavior should be if/when we get around to updating the rules to adhere to how we treat attributes in Candlepin. The rules treat null values the same as "not present," and they merge pool and product attributes in a prioritized method, depending on which you call. Since fixing all of that is a task in itself, I opted to just comment on the broken behavior and leave it.